packages/@angular/cli/tasks/npm-install.ts

```diff
@@ -11,22 +12,24 @@ export default Task.extend({
       packageManager = 'npm';
     }
 
-    return new Promise(function(resolve, reject) {
-      ui.writeLine(chalk.green(`Installing packages for tooling via ${packageManager}.`));
-      let installCommand = `${packageManager} install`;
-      if (packageManager === 'npm') {
-        installCommand = `${packageManager} --quiet install`;
-      }
-      exec(installCommand,
-        (err: NodeJS.ErrnoException, _stdout: string, stderr: string) => {
-        if (err) {
-          ui.writeLine(stderr);
-          ui.writeLine(chalk.red('Package install failed, see above.'));
-          reject();
-        } else {
-          ui.writeLine(chalk.green(`Installed packages for tooling via ${packageManager}.`));
-          resolve();
+    return checkYarnOrCNPM().then(function () {
+      return new Promise(function(resolve, reject) {
```

Comment:
@Brocco yep, seems that you are right, just want to clarify one moment, if we will do it in this way:

```
return checkYarnOrCNPM().then(function () {
      ui.writeLine(chalk.green(`Installing packages for tooling via ${packageManager}.`));
      let installCommand = `${packageManager} install`;
      if (packageManager === 'npm') {
        installCommand = `${packageManager} --quiet install`;
      }
      exec(installCommand,
        (err: NodeJS.ErrnoException, _stdout: string, stderr: string) => {
        if (err) {
          ui.writeLine(stderr);
          ui.writeLine(chalk.red('Package install failed, see above.'));
          throw new Error('Package install failed, see above.');
        } else {
          ui.writeLine(chalk.green(`Installed packages for tooling via ${packageManager}.`));
        }
      });
    });
```

then we will have kind off ugly output like this one:

```
Successfully initialized git.
Seems that yarn is not installed.
You can `ng set --global packageManager=npm`.
Installing packages for tooling via yarn.
Seems that yarn is not installed.
You can `ng set --global packageManager=npm`.
Project 't2' successfully created.
'yarn' is not recognized as an internal or external command,
operable program or batch file.

Package install failed, see above.
C:\Users\AlexandrM\Desktop\angular-cli\packages\@angular\cli\tasks\npm-install.ts:24
                    throw new Error('Package install failed, see above.');
                    ^

Error: Package install failed, see above.
    at child_process_1.exec (C:\Users\AlexandrM\Desktop\angular-cli\packages\@angular\cli\tasks\npm-install.ts:24:27)
    at ChildProcess.exithandler (child_process.js:218:5)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:885:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
```

versus:

```
Successfully initialized git.
Seems that yarn is not installed.
You can `ng set --global packageManager=npm`.
Installing packages for tooling via yarn.
'yarn' is not recognized as an internal or external command,
operable program or batch file.

Package install failed, see above.
```

plus possible future modifications may be harder, all we do in this commit is just wrapped one promise into another, later on, if there will be better decision all work need to be done is just unwrap this guy

* I am just asking, if you will say "just do it that way" it is absolutely fine